### ISSUE TITLE
GoogleSignIn().signIn() causes IPhone app to crash

### ISSUE THREAD

**Issue by JakubSochacki06** [flutter/flutter#150671]
### Steps to reproduce

1. Call the GoogleSignIn().signIn() function

### Expected results

Open popup so I can log in like on Android (It works completely fine on Android, it was working fine on iPhone aswell on the start, but at some point it stopped working)

### Actual results

App crash

### Code sample

<details open><summary>Code sample</summary>

```dart
  Future<void> signInWithGoogle(context) async {
    try {
      print("tried to googlesign in");
      final GoogleSignInAccount? googleUser = await GoogleSignIn().signIn();
      print("passed it!");
      if (googleUser == null) return;

      final GoogleSignInAuthentication googleAuth =
          await googleUser.authentication;
      final AuthCredential credential = GoogleAuthProvider.credential(
        accessToken: googleAuth.accessToken,
        idToken: googleAuth.idToken,
      );
      await _signInWithCredential(credential);
    } catch (e) {
      print(e);
      showModalBottomSheet(
        context: context,
        builder: (builder) {
          return Container(
            child: Text(e.toString()),
          );
        },
      );
    }
  }
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

https://github.com/flutter/flutter/assets/104648918/32885161-9e65-442e-a99d-bc9054c556bb

### Logs

<details open><summary>Logs</summary>

```console
flutter: tried to googlesign in
Lost connection to device.
Error: Unable to terminate com.sohatzky.outwork on C1EA8AEA-F758-4BCD-9A9B-A4928405B2EE:
ProcessException: Process exited abnormally with exit code 3:
An error was encountered processing the command (domain=NSPOSIXErrorDomain, code=3):
Simulator device failed to terminate com.sohatzky.outwork.
found nothing to terminate
Underlying error (domain=NSPOSIXErrorDomain, code=3):
	The request to terminate "com.sohatzky.outwork" failed. found nothing to terminate
	found nothing to terminate
  Command: /usr/bin/arch -arm64e xcrun simctl terminate C1EA8AEA-F758-4BCD-9A9B-A4928405B2EE com.sohatzky.outwork
the Dart compiler exited unexpectedly.
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.22.0, on macOS 14.5 23F79 darwin-arm64, locale
    en-PL)
[!] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.
[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
[✗] Chrome - develop for the web (Cannot find Chrome executable at
    /Applications/Google Chrome.app/Contents/MacOS/Google Chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.
[✓] Android Studio (version 2023.3)
[✓] Connected device (4 available)
[✓] Network resources

```

</details>

**Comment by huycozy** [comment]
Could you run with verbose (`-v`) to see if you can get the entire crash stack trace? If the output is the same as above, you can try debugging it from XCode. And please share the crash stack trace when you have it. Thanks!

**Comment by JakubSochacki06** [comment]
Hey, thank you for your reply.
I tried running flutter run -v, here is the video of me clicking the button again with verbose.

Crash text:
`[ +203 ms] The Flutter DevTools debugger and profiler on iPhone 15 Pro Max is available at: http://127.0.0.1:9101?uri=http://127.0.0.1:50469/VOo-cPCkwA4=/
[+16917 ms] Service protocol connection closed.
[        ] Lost connection to device.
[   +1 ms] DevFS: Deleting filesystem on the device
(file:{location}


)
[ +253 ms] Ignored error while cleaning up DevFS: TimeoutException after 0:00:00.250000: Future not completed

[   +2 ms] "flutter run" took 53,203ms.
[ +251 ms] ensureAnalyticsSent: 209ms
[        ] Running 1 shutdown hook
[   +8 ms] Shutdown hooks complete
[   +2 ms] exiting with code 0
`


https://github.com/flutter/flutter/assets/104648918/2885330e-e9b5-48cd-a829-6e569e6373b5

I don't see anything new here, could you please let me know what do you mean by saying that I can try to debug it with XCode? I don't really understand how I could do that.

**Comment by JakubSochacki06** [comment]
For future people trying to solve the problem:
At some point after pulling files from git my info.plist contained 2 CFBundleURLTypes - one that plugin required and one that just appeared. I had to delete the second one, here is how it looked like.

```
	<key>CFBundleURLTypes</key>
    <array>
      <dict>
        <key>CFBundleTypeRole</key>
        <string>Editor</string>
        <key>CFBundleURLSchemes</key>
        <array>
          <string>$(PRODUCT_BUNDLE_IDENTIFIER)</string>
        </array>
      </dict>
    </array>
```
